addons/l10n_vn_edi_viettel/models/account_move.py

```diff
@@ -0,0 +1,766 @@
+        self.ensure_one()
+
+        def grouping_key_generator(base_line, tax_values):
+            # Requirement is to generate a tax breakdown per taxPercentage
+            return {'tax_percentage': tax_values['tax_repartition_line'].tax_id.amount or -2}
+
+        tax_breakdowns = []
+
+        tax_details_grouped = self._prepare_invoice_aggregated_taxes(grouping_key_generator=grouping_key_generator)
+        for tax_percentage, tax_percentage_values in tax_details_grouped['tax_details'].items():
+            tax_breakdowns.append({
+                'taxPercentage': tax_percentage['tax_percentage'],
+                'taxableAmount': tax_percentage_values['base_amount_currency'],
+                'taxAmount': tax_percentage_values['tax_amount_currency'],
+                'taxableAmountPos': self.move_type == 'out_invoice',  # For adjustment invoice, the amount should be considered as negative.
+                'taxAmountPos': self.move_type == 'out_invoice',  # Same
+            })
+
+        json_values['taxBreakdowns'] = tax_breakdowns
+
+    def _l10n_vn_edi_lookup_invoice(self):
+        """ Lookup on invoice, returning its current details on SInvoice. """
+        self.ensure_one()
+        invoice_data, error_message = self._l10n_vn_edi_send_request(
+            method='POST',
+            endpoint='InvoiceAPI/InvoiceWS/searchInvoiceByTransactionUuid',
+            params={
+                'supplierTaxCode': self.company_id.vat,
+                'transactionUuid': self.l10n_vn_edi_invoice_transaction_id,
+            },
+        )
+        return invoice_data, error_message
+
+    @api.model
+    def _l10n_vn_edi_send_request(self, method, endpoint, json_data=None, params=None):
+        """ Send a request to the API based on the given parameters. In case of errors, the error message is returned. """
+        url = urllib.parse.urljoin(SINVOICE_API_URL, endpoint)
+        if json_data:
+            headers = {'Content-Type': 'application/json; charset=utf-8'}
+        else:
+            headers = {'Content-Type': 'application/x-www-form-urlencoded;'}
+            params = urllib.parse.urlencode(params)
+
+        if endpoint != '/auth/login':  # in which case we do not yet have the token.
+            headers['Cookie'] = f'access_token={self._l10n_vn_edi_get_access_token()}'
+
+        try:
+            response = requests.request(method, url, json=json_data, params=params, headers=headers, timeout=SINVOICE_TIMEOUT)
+            resp_json = response.json()
+            error = None
+            if resp_json.get('code') or resp_json.get('error'):
+                data = resp_json.get('data') or resp_json.get('error')
+                error = _('Error when contacting SInvoice: %s.', data)
+            return resp_json, error
+        except (RequestException, ValueError) as err:
+            raise UserError(_('Something went wrong, please try again later: %s', err))
```

Comment:
This method is called from account.move.send. You must not raise anything except if you explicitely catch the error somewhere. You must not rollback the entire wizard in that case. What about returning the error instead?